### ISSUE TITLE
runtime: merge .go files produced by cgo into their counterparts

### ISSUE THREAD

**Issue by nodirt** [golang/go#12952]
Merge 
- `panic1.go` -> `panic.go`
- `defs1_darwin.go` + `defs2_darwin.go` -> `defs_darwin.go`
- etc

Discussed in https://groups.google.com/forum/#!topic/golang-nuts/yuh2fzooEpM

I'd like to work on this.

**Comment by stemar94** [comment]
@nodirt What about runtime{1,2,}.go ?

**Comment by nodirt** [comment]
See comments in https://go-review.googlesource.com/#/c/16025/

**Comment by RLH** [comment]
In general this will cause a lot of our CLs to have conflicts and slow us
down as we approach the Nov. 1 code freeze. The upside seems minimal given
that any reasonable SDE will figure out where a function is regardless of
the file it is in.

**Comment by nodirt** [comment]
Yes, I was worried that it will interfere with your and @aclements CLs. I can postpone it.

I'm not sure how it slows you down though. https://go-review.googlesource.com/#/c/16008/ demonstrated that git-rebase handles non-invasive changes well (I don't do invasive ones)

**Comment by aclements** [comment]
It depends on how git sees the merge and rename. Typically, it will rebase changes to the largest hunk of a merged file without trouble, but it will have problems following changes to the smaller hunks over. I'll try to flip through your unmerged CLs later and give my opinion on which should be postponed. I definitely want to see this change and thank you for going to this trouble!

**Comment by aclements** [comment]
CL 16024 may cause problems, but I see it's already submitted, so we'll just have to see what happens. CL 16025 (runtime*.go) will definitely cause problems, so if you don't mind I'd like to hold off on that one until things have settled down a little (it looks like that one may take a little more thought, anyway). I don't think any of the other CLs will cause any problems.

**Comment by nodirt** [comment]
Yes, CL 16025 would cause problems for you. That's why I abandoned it with message "Rearranging runtime*.go is more invasive and requires some planning and coordination with runtime team. I will leave them for now.
It is not the only files I will leave. There is also unaligned*.go that have different build tags."

**Comment by mdempsky** [comment]
I'd like to see the os*_foo.go files collapsed into os_foo.go.  Any objections to that?

Rationale: Currently it's rather unfortunate that Gerrit sorts os1_*.go files together, then os2_*.go files, etc.  That makes it more annoying (at least IMO) to review CLs like https://golang.org/cl/16176, where it would be nicer to see the os_foo.go and os1_foo.go changes together.

As far as I can tell, CL 16176 is the only pending CL that touches the os*.go files, but I have a couple more similar cleanups in mind.
